extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridStringArrayConverter.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    /// <summary>
+    /// Converter to bind to <see cref="string[]" /> type parameters.
+    /// </summary>
+    [SupportedTargetType(typeof(string[]))]
+    internal class EventGridStringArrayConverter : EventGridConverterBase
+    {
+        protected override ConversionResult ConvertCore(Type targetType, string json)
+        {
+            if (targetType != typeof(string[]))
```

Comment:
Does this mean that if customer wants to use a collection type, it must be array? what happens if they use `IEnumerable<string>` ? Would there be another converter (built-in one may be ?) handles that?

Reply:
Yes, we are only supported arrays as the goal is parity with in-proc. We have an issue to track collection support improvements for all extensions https://github.com/Azure/azure-functions-dotnet-worker/issues/1334